inc/functions/custom-folders.php

```diff
@@ -597,3 +619,173 @@ function imagify_insert_custom_file( $args = array() ) {
+	$args = array_merge( array(
+		'file_id'     => 0,
+		'file_path'   => '',
+		'path'        => '',
+		'backup_path' => '',
+		'file'        => false,
+	), $args );
+
+	$filesystem = imagify_get_filesystem();
+	$file       = $args['file'] && is_a( $args['file'], 'Imagify_File_Attachment' ) ? $args['file'] : false;
+
+	// The file.
+	if ( ! $args['file_path'] && $args['path'] ) {
+		$args['file_path'] = Imagify_Files_Scan::remove_placeholder( $args['path'] );
+	}
+
+	if ( ! $args['file_path'] && $args['file_id'] ) {
+		$file = $file ? $file : get_imagify_attachment( 'File', $args['file_id'], 'delete_custom_file' );
+		$args['file_path'] = $file->get_original_path();
+	}
+
+	if ( $args['file_path'] && $filesystem->exists( $args['file_path'] ) ) {
+		$filesystem->delete( $args['file_path'] );
+	}
+
+	// The backup file.
+	if ( ! $args['backup_path'] && $args['file_path'] ) {
+		$args['backup_path'] = imagify_get_file_backup_path( $args['file_path'] );
+	}
+
+	if ( ! $args['backup_path'] && $args['file_id'] ) {
+		$file = $file ? $file : get_imagify_attachment( 'File', $args['file_id'], 'delete_custom_file' );
+		$args['backup_path'] = $file->get_raw_backup_path();
+	}
+
+	if ( $args['backup_path'] && $filesystem->exists( $args['backup_path'] ) ) {
+		$filesystem->delete( $args['backup_path'] );
+	}
+
+	// In the database.
+	if ( $file ) {
+		$file->delete_row();
+	} else {
+		Imagify_Files_DB::get_instance()->delete( $args['file_id'] );
+	}
+}
+
+/**
+ * Check if a file has been modified, and update the database accordingly.
+ *
+ * @since  1.7
+ * @author Grégory Viguier
+ *
+ * @param  object $file    An Imagify_File_Attachment object.
+ * @return int|bool|object The file ID if modified. False if not modified. A WP_Error object if the file doesn't exist.
+ */
+function imagify_refresh_file_modified( $file ) {
+	$file_path   = $file->get_original_path();
+	$backup_path = $file->get_backup_path();
+	$filesystem  = imagify_get_filesystem();
+	$modified    = false;
+
+	if ( ! $file_path || ! $filesystem->exists( $file_path ) ) {
+		/**
+		 * The file doesn't exist anymore.
+		 */
+		if ( $backup_path ) {
+			// Delete the backup file.
+			$filesystem->delete( $backup_path );
+		}
+
+		// Remove the entry from the database.
+		$file->delete_row();
+
+		return new WP_Error( 'no-file', __( 'The file was missing or its path couldn\'t be retrieved from the database. The entry has been deleted from the database.', 'imagify' ) );
+	}
+
+	/**
+	 * The file still exists.
+	 */
+	$old_data = $file->get_row();
+	$new_data = array();
+
+	// Folder ID.
+	if ( $old_data['folder_id'] ) {
+		$folder = Imagify_Folders_DB::get_instance()->get( $old_data['folder_id'] );
+
+		if ( ! $folder ) {
+			$new_data['folder_id'] = 0;
+		}
+	}
+
+	// Hash + modified.
+	$current_hash = md5_file( $file_path );
+
+	if ( ! $old_data['hash'] ) {
+		$new_data['modified'] = 0;
+	} else {
+		$new_data['modified'] = (int) ! hash_equals( $old_data['hash'], $current_hash );
```

Comment:
you have a compat for hash_equals? (since it's PHP 5.6)

Reply:
You're right, it was missing. The polyfill appeared in WP only in version 3.9.2.